### ISSUE TITLE
Add mondatory state keys to useStoreon

### ISSUE THREAD

**Issue by ai** [distolma/storeon-solidjs#1]
Storeon React `useStoreon` has arguments for a reason. Without it all your components will re-render on any state changes.

**Comment by distolma** [comment]
There are no reasons to provide technics from React to Solid. We use mandatory state keys to React `useStoreon` to avoid re-render due to React's Virtual DOM and specific works with hooks. In this case, Solid is more like Vue or Svelte, there is no Virtual DOM, and it uses a reactive graph to detect changes and re-render (https://github.com/ryansolid/solid/blob/master/documentation/comparison.md#react)

Here is a little comparison of how those frameworks render own component
React: https://codesandbox.io/s/magical-poitras-w7l18?file=/src/App.js:0-307
Solid: https://codesandbox.io/s/solid-counter-hn2cq?file=/index.tsx

**Comment by ai** [comment]
As I can see there are three layers in Svelte, React, or Solid.js:

1. Global state
2. Controller state
3. DOM

Svelte and Solid.js improved the connection between the Controller state and the DOM layers. But the problem between the Global state and Controller state is still actual.

For instance, here is Solid.js controller:

```js
export default function Counter() {
  const [state, dispatch] = useStoreon()

  return (
    <div>
      {state.count}
      <button onClick={() => dispatch('inc')}>inc</button>
    </div>
  )
}
```

Every time when you will change any state, this code will be executed:

```js
const [state, dispatch] = useStoreon()
```

In real apps, with 100+ components with a lot of hooks inside, these calls will slow down performance even outside of DOM operations.

**Comment by distolma** [comment]
This function will executed only once, on initial render.
```js
const [state, dispatch] = useStoreon()
```

The `state` is a proxied object of Storeon state and due to that Solid detect changes on the store directly, and if the property used somewhere, it will be reactively updated.

```js
const App = () => {
  const [state, dispatch] = useStoreon();

  const timer = setInterval(() => dispatch("inc"), 1000);
  onCleanup(() => clearInterval(timer));

  createEffect(() => {
    console.log(state.foo);
  });

  return <div>{state.count}</div>;
};

function inc(store) {
  store.on("@init", () => ({ count: 0, foo: "baz" }));
  store.on("inc", state => ({ count: state.count + 1 }));
}
const store = createStoreon([inc]);
```

The log function inside `createEffect` will be called only once. If we change `count`, there will no reaction to that, because we don't use it, but if we change `foo`, our log function will be called again.
For reference: https://codesandbox.io/s/solid-counter-hn2cq?file=/index.jsx:0-773

**Comment by ai** [comment]
What is I have code like:

```js
const App = () => {
  const [state, dispatch] = useStoreon();

  const users = state.admins.lenght + state.citizens.length;

  return <div>{users}</div>;
};
```

When `state.admins.lenght + state.citizens.length` will be called on `state.admins` changes?

**Comment by distolma** [comment]
No, exactly this code will not work, because when you add two of these properties (`admins.length`, `citizens.length`) you lose reactive reference of parent object `state`.

But if you put computation into tempalte, like
```js
const App = () => {
  const [state, dispatch] = useStoreon();

  return <div>{state.admins.lenght + state.citizens.length}</div>;
};
```
or use `createMemo` function from `solid-js`

```js
const App = () => {
  const [state, dispatch] = useStoreon();

  const users = createMemo(() => state.admins.lenght + state.citizens.length);

  return <div>{users}</div>;
};
```

or even like this, with component state

```js
const App = () => {
  const [innerState, setInnerState] = createState({ users: 0 });
  const [state, dispatch] = useStoreon();

  createEffect(() => {
    const users = state.admins.lenght + state.citizens.length;
    setInnerState({ users })
  })

  return <div>{innerState.users}</div>;
};
```
it will work)

**Comment by ai** [comment]
Got it.

**Comment by ai** [comment]
Move the repo to `storeon` org and release it under `@storeon/solidjs` name. Ping me after each step, I will fix rights.

**Comment by ai** [comment]
If you want to promote Solid.js framework, feel free to write a tweet about `@storeon/solidjs` with a quick description of what is Storeon (Redux-like state manager in X bytes) and Solid.js. It will be nice to have component’s source code and cleaned compiled code images (to show benefits of Solid.js in image).

I will report this tweet, since I like any Svelte-like project.

**Comment by distolma** [comment]
@ai Package is moved
https://www.npmjs.com/package/@storeon/solidjs

Please rename this repo to `solidjs`
